### PR TITLE
3 inventory items sprint 2

### LINKED ISSUE #121: Updated Inventory UI
# **Updated Inventory UI**
**Task:** [Updated Inventory UI].
**Feature:** [Inventory/items](#3)

This task is for implementing and updating features on the inventory and item user interface. For example the inventory buttons need to be updated and inventory item pop up descriptions need to be implemented.

# **Milestones**
List of steps that need to be completed for this task.
- [x] Have pop up implemented (Aug. 26)
- [x] Have Item pop up dispose when item clicked (Aug 28)
- [ ] Main pull request (Aug. 28)

**Completion Deadline:** Aug. 28

# **Documentation**
- [Main description of feature](../wiki/Game-Screens)
- [JavaDoc](JavaDoc/Location/Package/Class)
- ...
 
# **Member**
- Tom (@tompreston1) `tom152738379`

### LINKED ISSUE #164: Random Spawning of Items - Integrated into Chunk Generation
# **Description**
**Task:** [Task Name](#3).
**Feature:** [Team 3 Sprint 2 Features](#157 )

Item spawning needs to be integrated into chunk generation to only spawn items based on a) time and b) player location (ie items should only be spawned on chunks nearby the player, and disposed of if far from the player to avoid unnecessary lag).

# **Dependencies**
 
# **Milestones**
List of steps that need to be completed for this task. At the end of each step, tests must be written
- [ ] Set up system for spawning items randomly within a chunk (Sep 3)
- [ ] Enable chunks around player to spawn items dynamically (Sep 3)
- [ ] Dispose of items far from the player (ie, more than a certain number of chunks away) (Sep 4)
- [ ] Enable timed spawning of items randomly based on some probability distribution. (Sep 4)
- [ ] Check all tests (should be written with each step) (Sep 5)
- [ ] Tweak configuration files for probability distributions per tile for spawn rates. (Sep 5)
- [ ] Merge into team 3 branch (Sep 5)
- [ ] Create release version to be merged into main (Sep 5)
- [ ] Merge into main (Sep 6)
 
**Completion Deadline:** Sep. 7

 
# **Documentation**
- TBC

 
# **Member**
- Mittun (@mitsudz) `mitsudz`
- Howie (@Howard-12)
- Praneet (@praneetdhoolia)
- TEAM 8 MEMBER TBC

### PR DESCRIPTION
**Pull request by mitsudz**
First team 3 pr to main

Adds in random generation of items integrated with newly refactored chunk generation. Also disposes of far away items.
Adds in dialogue box when hovering over an item in the inventory.

Closes #164 
Closes #214 
Closes #121

### CODE REVIEW

**Review by praneetdhoolia — APPROVED** [review]
Looks good to go. Great work!

File: source/core/src/main/com/csse3200/game/areas/terrain/TerrainComponent.java
Comment (BaileyS03) [inline source/core/src/main/com/csse3200/game/areas/terrain/TerrainComponent.java:1]: 0% test coverage -- idk if this is your problem though

File: source/core/src/main/com/csse3200/game/areas/terrain/TerrainLoader.java
Comment (BaileyS03) [inline source/core/src/main/com/csse3200/game/areas/terrain/TerrainLoader.java:1]: 0% test coverage -- idk if this is your problem though

File: source/core/src/main/com/csse3200/game/components/player/PlayerInventoryDisplay.java
Comment (BaileyS03) [inline source/core/src/main/com/csse3200/game/components/player/PlayerInventoryDisplay.java:1]: minimal testing here as well -- would like to see more in the future but can visually see it works in game play

**Review by BaileyS03 — COMMENTED** [review]
Please remove the file that is commented out or provide a reason and I will approve.

File: source/core/src/main/com/csse3200/game/areas/terrain/TerrainComponent.java
Comment (praneetdhoolia) [reply source/core/src/main/com/csse3200/game/areas/terrain/TerrainComponent.java:1]: https://github.com/UQcsse3200/2024-studio-2/issues/161

File: source/core/src/main/com/csse3200/game/areas/terrain/TerrainLoader.java
Comment (praneetdhoolia) [reply source/core/src/main/com/csse3200/game/areas/terrain/TerrainLoader.java:1]: TBD: Team 8

File: source/core/src/main/com/csse3200/game/components/minigames/MinigameCamera.java
[the remark this answers is not in the document]
Comment (mitsudz) [reply source/core/src/main/com/csse3200/game/components/minigames/MinigameCamera.java:1]: Have removed file - i think someone else had removed it from main (I assume Team 7) and it had just crept into the pr during some rebase.

File: .gitignore
Comment (Howard-12) [inline .gitignore:1]: You can have a `.gitignore` for each directory. Add `*` at `saves/.gitignore` should work.

**Review by Dabbyyyy — APPROVED** [review]
changes look good.

**Review by Howard-12 — APPROVED** [review]

File: .gitignore
Comment (mitsudz) [reply .gitignore:1]: I'll let someone else add that in another pr. Not sure why the overall .gitignore with /saves/* doesn't work tho.

**Pull request merged**